src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -205,19 +204,9 @@ public static TRoot InsertNodesAfter<TRoot>(this TRoot root, SyntaxNode nodeInLi
 
     public static string GetContent<TNode>(this TNode node) where TNode : SyntaxNode
     {
-        var builder = new StringBuilder();
-        foreach (var token in node.DescendantNodesAndSelf())
-        {
-            if (!token.IsToken)
-            {
-                continue;
-            }
-
-            var syntaxToken = (SyntaxToken)token;
-            builder.Append(syntaxToken.Content);
-        }
-
-        return builder.ToString();
+        using var writer = new System.IO.StringWriter();
```

Comment:
@DustinCampbell when you're back from vacation, are you planning on putting the pooled StringWriter stuff in our solution? If so, this is a good candidate.

Reply:
Unless "the pooled StringWriter stuff" means something specific to you and Dustin, which wouldn't surprise me, the object pooling infrastructure in general is in main, so a StringWriterPool should be easy to create, presumably following the pattern of https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.AspNetCore.Razor.Common/PooledObjects/StringBuilderPool.cs

Reply:
Instead of adding `StringWriterPool`, I used the existing `StringBuilderPool` - a pattern I found in Roslyn and makes sense to me (also not sure how would I write `StringWriterPool.Policy`).

However, the object pools are not visible in this project, so I needed to add bunch of references. But it broke source build :(

Reply:
I would avoid adding new references to the project for this specific change. If we can take advantage of pool without changes then lets do so, otherwise let's skip.

My reasoning for this is @DustinCampbell is working on cleaning up our string pooling infrastructure in Razor. That will allow us to share pools across the IDE and compiler team. I don't want to potentially make his change harder by introducing references he may or may not be trying to actively remove.

Reply:
Agreed Jared. Let's leave this as is for now: one StringWriter allocation per call is _much_ better than the linq allocations that we did previously. We can improve more later 🙂.

Reply:
Makes sense, reverted the last commit.